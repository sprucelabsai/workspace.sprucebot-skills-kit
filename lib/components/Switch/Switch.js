'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Switch = function (_Component) {
	_inherits(Switch, _Component);

	function Switch(props) {
		_classCallCheck(this, Switch);

		var _this = _possibleConstructorReturn(this, (Switch.__proto__ || Object.getPrototypeOf(Switch)).call(this, props));

		_this.state = {
			on: !!props.on
		};
		return _this;
	}

	_createClass(Switch, [{
		key: 'onChange',
		value: function onChange(e) {
			var _this2 = this;

			// toggle on state of button
			this.setState(function (prevState, props) {
				// new 'on' state
				var on = !prevState.on;

				// let any callbacks know of state change
				if (_this2.props.onChange) {
					_this2.props.onChange(on, e);
				}

				return {
					on: on
				};
			});
		}
	}, {
		key: 'render',
		value: function render() {
			var props = Object.assign({}, this.props);

			var className = (props.className || '') + ' switch';

			delete props.className;
			delete props.on;
			delete props.onChange;
			delete props.children;

			// are we on?
			if (this.state.on) {
				className += ' on';
			}

			return _react2.default.createElement(
				'div',
				_extends({
					className: className
				}, props, {
					onMouseUp: this.onChange.bind(this)
				}),
				_react2.default.createElement('button', { type: 'button' })
			);
		}
	}]);

	return Switch;
}(_react.Component);

exports.default = Switch;


Switch.propTypes = {
	on: _propTypes2.default.bool,
	onChange: _propTypes2.default.func
};

Switch.defaultProps = {
	on: false,
	onChange: function onChange(e) {}
};