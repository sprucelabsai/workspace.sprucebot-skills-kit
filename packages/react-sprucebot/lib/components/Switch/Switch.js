"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var Switch =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Switch, _Component);

  function Switch(props) {
    var _this;

    (0, _classCallCheck2.default)(this, Switch);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Switch).call(this, props));
    _this.state = {
      on: !!props.on
    };
    _this.onChange = _this.onChange.bind((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)));
    return _this;
  }

  (0, _createClass2.default)(Switch, [{
    key: "onChange",
    value: function onChange(e) {
      var _this2 = this;

      // toggle on state of button
      this.setState(function (prevState, props) {
        // new 'on' state
        var on = !prevState.on; // let any callbacks know of state change

        if (_this2.props.onChange) {
          _this2.props.onChange(on, e);
        }

        return {
          on: on
        };
      });
    }
  }, {
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      this.setState({
        on: nextProps.on
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this3 = this;

      return _react.default.createElement("div", {
        className: "switch switch".concat(this.props.on ? ' on' : '')
      }, _react.default.createElement("button", {
        onClick: function onClick() {
          return _this3.onChange();
        },
        type: "button"
      }, _react.default.createElement("span", null, this.props.on ? 'Enabled' : 'Disabled')));
    }
  }]);
  return Switch;
}(_react.Component);

exports.default = Switch;
Switch.propTypes = {
  on: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
Switch.defaultProps = {
  on: false
};