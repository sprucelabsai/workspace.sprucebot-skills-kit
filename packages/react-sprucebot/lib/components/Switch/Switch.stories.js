'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Switch = require('./Switch');

var _Switch2 = _interopRequireDefault(_Switch);

var _Switch3 = require('./Switch.md');

var _Switch4 = _interopRequireDefault(_Switch3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Switch', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_Switch4.default, function () {
  return _react2.default.createElement(_Switch2.default, null);
}));