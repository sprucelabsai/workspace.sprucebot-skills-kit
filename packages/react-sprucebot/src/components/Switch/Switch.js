import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Switch extends Component {
	constructor(props) {
		super(props)
		this.state = {
			on: !!props.on
		}

		this.onChange = this.onChange.bind(this)
	}
	onChange(e) {
		// toggle on state of button
		this.setState((prevState, props) => {
			// new 'on' state
			let on = !prevState.on

			// let any callbacks know of state change
			if (this.props.onChange) {
				this.props.onChange(on, e)
			}

			return {
				on: on
			}
		})
	}

	componentWillReceiveProps(nextProps) {
		this.setState({ on: nextProps.on })
	}

	render() {
		return (
			<div className={`switch switch${this.props.on ? ' on' : ''}`}>
				<button onClick={() => this.onChange()}>
					<span>{this.props.on ? 'Enabled' : 'Disabled'}</span>
				</button>
			</div>
		)
	}
}

Switch.propTypes = {
	on: PropTypes.bool,
	onChange: PropTypes.func
}

Switch.defaultProps = {
	on: false
}
