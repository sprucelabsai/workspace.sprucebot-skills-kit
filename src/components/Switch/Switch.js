import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Switch extends Component {
	constructor(props) {
		super(props)
		this.state = {
			on: !!props.on
		}
	}
	onChange(e) {
		// toggle on state of button
		this.setState((prevState, props) => {
			// new 'on' state
			let on = !prevState.on

			// let any callbacks know of state change
			if (this.props.onChange) {
				this.props.onChange(on, e)
			}

			return {
				on: on
			}
		})
	}
	render() {
		const props = Object.assign({}, this.props)

		let className = (props.className || '') + ' switch'

		delete props.className
		delete props.on
		delete props.onChange
		delete props.children

		// are we on?
		if (this.state.on) {
			className += ' on'
		}

		return (
			<div
				className={className}
				{...props}
				onMouseUp={this.onChange.bind(this)}
			>
				<button type="button" />
			</div>
		)
	}
}

Switch.propTypes = {
	on: PropTypes.bool,
	onChange: PropTypes.func
}

Switch.defaultProps = {
	on: false,
	onChange: e => {}
}
