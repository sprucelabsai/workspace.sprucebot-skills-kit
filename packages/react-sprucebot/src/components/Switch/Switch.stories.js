import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import Switch from './Switch'
import readme from './Switch.md'

const stories = storiesOf('Switch', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, () => <Switch on={boolean('On', false)} />)
)
