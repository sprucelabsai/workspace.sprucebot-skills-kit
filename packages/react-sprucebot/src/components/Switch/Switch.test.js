import React from 'react'
import renderer from 'react-test-renderer'
// setup file
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })
import { mount, shallow } from 'enzyme'

import 'jest-styled-components'
import Switch from './Switch'

test('it renders', () => {
	const tree = renderer.create(<Switch />).toJSON()
	expect(tree).toMatchSnapshot()
})
test('it should respect props', () => {
	const wrapper = shallow(<Switch on={true} />, { lifecycleExperimental: true })
	expect(wrapper.instance().props.on).toEqual(true)
	wrapper.setProps({ on: false })
	expect(wrapper.instance().props.on).toEqual(false)
})

test('it should trigger prop changes', () => {
	const wrapper = mount(<Switch on={false} />, {
		lifecycleExperimental: true
	})
	expect(wrapper.instance().props.on).toEqual(false)
	wrapper.setProps({ on: true })
	expect(wrapper.instance().props.on).toEqual(true)
})

test('it set state onChange', () => {
	const wrapper = shallow(<Switch on={true} />).instance()
	let foo = wrapper.onChange(false)
	expect(wrapper.state.on).toEqual(false)
})

test('it set trigger onChange that is passed in', () => {
	var spy = jest.fn()
	const wrapper = shallow(<Switch onChange={spy} />).instance()
	let foo = wrapper.onChange(true)
	expect(wrapper.state.on).toEqual(true)
	expect(spy).toHaveBeenCalledTimes(1)
})
