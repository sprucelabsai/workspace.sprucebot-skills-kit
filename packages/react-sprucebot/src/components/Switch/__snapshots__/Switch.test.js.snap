// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="switch switch"
>
  <button
    onClick={[Function]}
  >
    <span>
      Disabled
    </span>
  </button>
</div>
`;
